homeassistant/components/climate/intesishome.py

```diff
@@ -0,0 +1,350 @@
+"""
+Support for IntesisHome Smart AC Controllers.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/climate.intesishome/
+"""
+
+import logging
+from datetime import timedelta
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import (CONF_PASSWORD, CONF_USERNAME, CONF_STRUCTURE)
+from homeassistant.util import Throttle
+from homeassistant.components import persistent_notification
+from homeassistant.components.climate import (ClimateDevice,
+                                              PLATFORM_SCHEMA,
+                                              ATTR_TEMPERATURE,
+                                              ATTR_OPERATION_MODE)
+from homeassistant.const import (TEMP_CELSIUS, STATE_UNKNOWN)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'IntesisHome'
+REQUIREMENTS = ['pyintesishome==0.4']
+STATE_FAN = 'Fan'
+STATE_HEAT = 'Heat'
+STATE_COOL = 'Cool'
+STATE_DRY = 'Dry'
+STATE_AUTO = 'Auto'
+STATE_QUIET = 'Quiet'
+STATE_LOW = 'Low'
+STATE_MEDIUM = 'Medium'
+STATE_HIGH = 'High'
+STATE_OFF = 'Off'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_STRUCTURE): vol.All(cv.ensure_list, cv.string),
+})
+
+# Return cached results if last scan time was less than this value.
+# If a persistent connection is established for the controller,
+# changes to values are in realtime.
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=180)
+
+try:
+    # pylint: disable=unused-import
+    from asyncio import ensure_future
+except ImportError:
+    # Python 3.4.3 and ealier has this as async
+    # pylint: disable=unused-import
+    from asyncio import async
+    ensure_future = async
+
+INTESISHOME = None
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the IntesisHome interface."""
+    from pyintesishome import IntesisHome
+    global INTESISHOME
```

Comment:
Thanks for your review Fabian, I've now switched the platform over to using hass.data